### PR TITLE
Add VPA objects for control plane components

### PR DESCRIPTION
**Pull request by wyb1**
**What this PR does / why we need it**:
Adds VPA objects for most of the control plane components.  Will be useful for cost analysis, to see how many resources each component really uses. VPA objects are all in `Off` mode so there will be no disruption.
/cc @mliepold 
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
Kube-apiserver was excluded because the VPA should not be used with an object that uses an HPA based on cpu or memory metrics.
Some other components were also excluded because they are not deployed in every control plane (Infrastructure specific or behind feature gate).

**Release note**:

```improvement operator
Added VPA objects for control plane components. VPA objects are in `Off` mode.
```

### CODE REVIEW

File: charts/seed-controlplane/charts/kube-controller-manager/templates/mcm-vpa.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{{ if .Values.vpa.enabled }}
```
Comment (rfranzke) [inline charts/seed-controlplane/charts/kube-controller-manager/templates/mcm-vpa.yaml]: Rename file to kube-controller-manager-vpa.yaml

**Review by rfranzke — COMMENTED** [review]
Looks good, just a small nit.

File: charts/seed-controlplane/charts/kube-controller-manager/templates/mcm-vpa.yaml (hunk 1 above)
Comment (wyb1) [reply charts/seed-controlplane/charts/kube-controller-manager/templates/mcm-vpa.yaml]: Fixed

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**